src/mpid/ch3/channels/nemesis/include/mpid_nem_datatypes.h

```diff
@@ -183,10 +175,7 @@ MPID_nem_cell_rel_ptr_t;
 typedef struct MPID_nem_cell
 {
     MPID_nem_cell_rel_ptr_t next;
-#if (MPID_NEM_CELL_HEAD_LEN > SIZEOF_MPL_ATOMIC_PTR_T)
-    char padding[MPID_NEM_CELL_HEAD_LEN - sizeof(MPID_nem_cell_rel_ptr_t)];
-#endif
-    volatile MPID_nem_pkt_t pkt;
+    volatile MPID_nem_pkt_t pkt MPL_ATTR_ALIGNED(8);
```

Comment:
I presume if alignment is not supported by the compiler, the assertion added will fail?

Reply:
You mean this assertion:
```
MPIR_Assert(MPID_NEM_ALIGNED(&((MPID_nem_cell_t*)0)->pkt.payload[0], 8));
```
Yeah, as long as this assertion is what we really intended, then it will fail when it fails. However, I guess I need to double-check why this alignment is important in nemesis. 

Actually, if we only want to ensure the alignment of `payload`, the attr alignment here probably is unnecessary. It is only necessary in the old code because we are manually padding it.